packages/yarnpkg-pnp/sources/loader/makeApi.ts

```diff
@@ -819,7 +819,18 @@ export function makeApi(runtimeState: RuntimeState, opts: MakeApiOptions): PnpAp
       const containingPackage = findPackageLocator(unqualifiedPath);
       if (containingPackage) {
         const {packageLocation} = getPackageInformationSafe(containingPackage);
-        if (!opts.fakeFs.existsSync(packageLocation)) {
+
+        let stat: Stats | undefined;
+        try {
+          stat = opts.fakeFs.statSync(packageLocation);
```

Comment:
Since we're not really using the `Stats` data we could use `accessSync` instead to avoid unnecessary allocations